Tests/Models/User.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Part of the Joomla GSoC Webservices Project
+ *
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+
+namespace Joomla\Entity\Tests\Models;
+
+/**
+ * Class User
+ * @package Joomla\Entity\Tests
+ * @since 1.0
+ */
+class User extends \Joomla\Entity\Model
+{
+	/**
+	 * @var array
+	 */
+	protected $defaultParams = array(
```

Comment:
Rather than `defaultParams` use a mutator for this - or the built in json support you've done :)